src/main/scala/zio/actors/ActorRef.scala

```diff
@@ -0,0 +1,133 @@
+package zio.actors
+
+import java.io.{ IOException, ObjectInputStream, ObjectOutputStream, ObjectStreamException }
+
+import zio.nio.{ InetAddress, InetSocketAddress, SocketAddress }
+import zio.nio.channels.AsynchronousSocketChannel
+import zio.{ DefaultRuntime, IO, Task, UIO }
+
+/**
+ *
+ * Reference to actor that might reside on local JVM instance or be available via remote communication
+ *
+ * @tparam E error type
+ * @tparam F wrapper type constructing DSL
+ */
+sealed trait ActorRef[E <: Throwable, -F[+_]] extends Serializable {
+
+  /**
+   *
+   * Send a message to actor - it can perform `fire-and-forget` interaction pattern or `ask` pattern
+   *
+   * @param fa message
+   * @tparam A return type
+   * @return effectful response
+   */
+  def ![A](fa: F[A]): Task[A]
+
+  /**
+   * Get referential absolute actor path
+   * @return
+   */
+  def path: UIO[String]
+
+}
+
+/* INTERNAL API */
+
+private[actors] object ActorRefSerial {
+
+  private[actors] val runtimeForResolve = new DefaultRuntime {}
+
+}
+
+private[actors] sealed abstract class ActorRefSerial[E <: Throwable, -F[+_]](private var actorPath: String)
+    extends ActorRef[E, F]
+    with Serializable {
+
+  import ActorSystemUtils._
+
+  @throws[IOException]
+  protected def writeObject1(out: ObjectOutputStream): Unit =
+    out.writeObject(actorPath)
```

Comment:
If this is throwing IOException, then maybe `IO.effect(out.writeObject(actorPath)).refineToOrDie[IOException]` would be appropriate (instead of the annotation)

Reply:
I've checked it and it turns out that `writeObject` and `readObject` needn't have these `@throws` annotations. Remaining signature is sufficient to override underlying Java serialization. If the annotations here are not informative, should I remove them? 

Also unfortunately I can't wrap inside of these functions in effect like that - Java serialization API enforces to do it by side effects (return type must be `Unit`) so if I wrap them in effect I also need to do it inside `runtime.unsafeRun` so the effect could be actually executed.
But such solution looks ugly to me and is not really useful: (so I would leave it as it is)
```scala
protected def writeObject1(out: ObjectOutputStream): Unit = 
  runtime.unsafeRun(IO.effect(out.writeObject(actorPath)).refineToOrDie[IOException])
```
What do you think?

Reply:
Ah, this is the grim part you mentioned during hackathon. I kinda share the reasoning, it is something imposed by the serialization API, I wouldn't even bother pushing for ZIO here. The levels relying it would have probably look like something both of you listed.